test_launch_ros/test/test_launch_ros/test_track_node_names.py

```diff
@@ -0,0 +1,161 @@
+    ls.include_launch_description(launch_description)
+    launch_task = loop.create_task(ls.run_async())
+    loop.run_until_complete(asyncio.sleep(5, loop=loop))
+    if not launch_task.done():
+        loop.create_task(ls.shutdown())
+        loop.run_until_complete(launch_task)
+    rclpy.shutdown()
+    return ls.context
+
+
+def test_launch_node_with_name():
+    node = Node(
+        package='demo_nodes_py',
+        node_executable='listener_qos',
+        node_name=TEST_NODE_NAME,
+        node_namespace='',
+        output='screen',
+    )
+    ld = LaunchDescription([node])
+    context = _launch(ld)
+    assert get_node_name_count(context, '/{}'.format(TEST_NODE_NAME)) == 1
+
+
+def test_launch_node_without_name():
+    node = Node(
+        package='demo_nodes_py',
+        node_executable='listener_qos',
+        node_namespace='',
+        output='screen',
+    )
+    ld = LaunchDescription([node])
+    context = _launch(ld)
+    assert get_node_name_count(context, '/{}'.format(TEST_NODE_NAME)) == 0
+
+
+def test_launch_composable_node_with_names():
+    node = ComposableNodeContainer(
+        package='rclcpp_components',
+        node_executable='component_container',
+        node_name=TEST_NODE_NAME,
+        node_namespace='',
+        composable_node_descriptions=[
+            ComposableNode(
+                package='composition',
+                node_plugin='composition::Listener',
+                node_name=TEST_NODE_NAME
+            )
+        ],
+        output='screen'
+    )
+    ld = LaunchDescription([node])
+    context = _launch(ld)
+    assert get_node_name_count(context, '/{}'.format(TEST_NODE_NAME)) == 2
+
+
+def test_launch_composable_node_without_component_name():
+    node = ComposableNodeContainer(
+        package='rclcpp_components',
+        node_executable='component_container',
+        node_name=TEST_NODE_NAME,
+        node_namespace='',
+        composable_node_descriptions=[
+            ComposableNode(
+                package='composition',
+                node_plugin='composition::Listener',
+            )
+        ],
+        output='screen'
+    )
+    ld = LaunchDescription([node])
+    context = _launch(ld)
+    assert get_node_name_count(context, '/{}'.format(TEST_NODE_NAME)) == 1
+    assert get_node_name_count(context, '/listener') == 1
+
+
+def test_launch_nodes_with_same_names():
+    node1 = Node(
+        package='demo_nodes_py',
+        node_executable='listener_qos',
+        node_name=TEST_NODE_NAME,
+        node_namespace='',
+        output='screen',
+    )
+
+    node2 = LifecycleNode(
+        package='lifecycle',
+        node_executable='lifecycle_listener',
+        node_name=TEST_NODE_NAME,
+        node_namespace='',
+        output='screen',
+    )
+
+    node3 = ComposableNodeContainer(
+        package='rclcpp_components',
+        node_executable='component_container',
+        node_name=TEST_NODE_NAME,
+        node_namespace='',
```

Comment:
It would be good to make use of the namespace too, so we can check that the fqn is being used.
It would also be good to check what happens when `node_name` is provided and no `node_namespace` (specially for `Node` action, where you can't know the fqn if both weren't provided in launch).

Reply:
Done.